File: Samples/Docking/VB/DockSample/Themes.vb
Imports System
Imports System.Diagnostics
Imports System.Reflection
Imports System.Windows

Namespace DevZest.DockSample

    Public Class Themes

        Private Shared s_currentTheme As ResourceDictionary = Nothing

        Public Overloads Shared Sub Load(ByVal themeName As String)
            Dim themeSource As Uri = New Uri(("/" _
                            + (Assembly.GetExecutingAssembly.FullName.Split(","c)(0) + (";component/themes/" _
                            + (themeName + ".xaml")))), UriKind.RelativeOrAbsolute)
            Load(themeSource)
        End Sub

        Public Overloads Shared Sub Load(ByVal themeSource As Uri)
            Reset()
            Debug.Assert((s_currentTheme Is Nothing))
            s_currentTheme = New ResourceDictionary
            s_currentTheme.Source = themeSource
            Application.Current.Resources.MergedDictionaries.Add(s_currentTheme)
        End Sub

        Public Shared Sub Reset()
            If (Not (s_currentTheme) Is Nothing) Then
                Application.Current.Resources.MergedDictionaries.Remove(s_currentTheme)
                s_currentTheme = Nothing
            End If
        End Sub
    End Class
End Namespace

File: Samples/Docking/CSharp/Themes/Themes.cs
using System;
using System.Diagnostics;
using System.Reflection;
using System.Windows;

namespace DevZest.Windows.Docking
{
    public static class Themes
    {
        static ResourceDictionary s_currentTheme = null;

        public static void Load(string themeName)
        {
            Uri themeSource = new Uri("/" + Assembly.GetExecutingAssembly().FullName.Split(',')[0] + ";component/themes/" + themeName + ".xaml", UriKind.RelativeOrAbsolute);
            Load(themeSource);
        }

        public static void Load(Uri themeSource)
        {
            Reset();
            Debug.Assert(s_currentTheme == null);
            s_currentTheme = new ResourceDictionary();
            s_currentTheme.Source = themeSource;
            Application.Current.Resources.MergedDictionaries.Add(s_currentTheme);
        }

        public static void Reset()
        {
            if (s_currentTheme != null)
            {
                Application.Current.Resources.MergedDictionaries.Remove(s_currentTheme);
                s_currentTheme = null;
            }
        }
    }
}
